src/include/platform/DeviceInfoProvider.h

```diff
@@ -92,9 +92,33 @@ class DeviceInfoProvider
     virtual UserLabelIterator * IterateUserLabel(EndpointId endpoint)   = 0;
 
 protected:
+    /**
+     * @brief Set the UserLabel at the specified index of the UserLabelList on a given endpoint
+     *
+     * @param endpoint - id to UserLabelList on which to set the UserLabel.
+     * @param index - index within the UserLabelList for which to set the UserLabel.
```

Comment:
How does this behave if the index is out of range for the current list length?  For example, what happens if the current length is 0 and the index passed here is 2?

Reply:
Rename SetUserLabelCount/GetUserLabelCount to SetUserLabelLength/GetUserLabelLength for better clarity. The API shall return CHIP_ERROR_INVALID_KEY_ID if specified index is out of range (0 ~ total length - 1)